tests/console/commands/test_add.py

```diff
@@ -125,7 +127,7 @@ def test_add_constraint_with_extras(app, repo, tester):
     assert 2 == tester._command.installer.executor.installations_count
 
 
-def test_add_constraint_dependencies(app, repo, tester):
+def test_add_constraint_dependencies(repo, tester):
     cachy2 = get_package("cachy", "0.2.0")
     msgpack_dep = get_dependency("msgpack-python", ">=0.5 <0.6")
     cachy2.requires = [msgpack_dep]
```

Comment:
This is the root of the problem with the test failures I was seeing.

When I put this in the fixture:

```
msgpack_dep = get_dependency("msgpack-python", ">=0.5 <0.6", optional=True)

cachy010 = get_package("cachy", "0.1.0")
cachy010.requires = [msgpack_dep]
repo.add_package(cachy010)

cachy020 = get_package("cachy", "0.2.0")
cachy020.requires = [msgpack_dep]
repo.add_package(cachy020)
...
```

Tests like `test_add_constraint_with_extras` fail with:

```
E       AssertionError: assert '\nUpdating d...chy (0.1.0)\n' == '\nUpdating d...chy (0.1.0)\n'
E         Skipping 79 identical leading characters in diff, use -v to show
E         - erations: 1 install, 0 updates, 0 removals
E         ?           ^
E         + erations: 2 installs, 0 updates, 0 removals
E         ?           ^        +
E           
E         +   • Installing msgpack-python (0.5.3)...
E         
E         ...Full output truncated (2 lines hidden), use '-vv' to show
```

and in the fixture if I change msgpak_dep to:

```
msgpack_dep = get_dependency("msgpack-python", ">=0.5 <0.6")
```

I get a failure of:

```
E           poetry.repositories.exceptions.PackageNotFound: Package [msgpack-python] not found.
```
I wasn't sure of a way to fix this problem. Any suggestions?

Reply:
That failure is expected, and is probably more representative of scenarios since you added msgpack as a dependency, poetry installs it.

For the failure, you just need to add a `msgpack-python` package with a compatible version to the repo.

Reply:
I think I am getting closer now. The fixture currently looks like:

```
@pytest.fixture(autouse=True)
def repo_add_default_packages(repo):
    repo.add_package(get_package("msgpack-python", "0.5.6"))
    msgpack_dep = get_dependency("msgpack-python", ">=0.5 <0.6", optional=True)

    cachy010 = get_package("cachy", "0.1.0")
    cachy010.extras = {"msgpack": [get_dependency("msgpack-python")]}
    cachy010.requires = [msgpack_dep]
    repo.add_package(cachy010)

    cachy020 = get_package("cachy", "0.2.0")
    cachy020.extras = {"msgpack": [get_dependency("msgpack-python")]}
    cachy020.requires = [msgpack_dep]
    repo.add_package(cachy020)

    repo.add_package(get_package("pendulum", "1.4.4"))
    repo.add_package(get_package("cleo", "0.6.5"))
    repo.add_package(get_package("tomlkit", "0.5.5"))
    repo.add_package(get_package("pyyaml", "3.13"))
    repo.add_package(get_package("pyyaml", "4.2b2"))
```

The issue I am having now is with cachy020 some of the tests expect msgpack-python to be required and some don't. So if i make it optional the ones that require it fail, but if I require it the ones that don't fail. For example `test_add_constraint_dependencies` fails when msgpack-python is optional, and `test_add_no_constraint` fails when it is not optional.

Reply:
You can modify the tests to make it work. In cases where adding the dependency means that the output now installs 2 packages (as it is also now installing the dependency), it is okay top update the output as long as it is not doing somehting unexpected.

Reply:
Alright, I was able to get all of the tests passing using the autouse.